src/Sylius/Bundle/ReviewBundle/DependencyInjection/SyliusReviewExtension.php

```diff
@@ -0,0 +1,115 @@
+     */
+    public function load(array $config, ContainerBuilder $container)
+    {
+        $config = $this->processConfiguration(new Configuration(), $config);
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+
+        $this->registerResources('sylius', $config['driver'], $this->resolveResources($config['resources'], $container), $container);
+
+        foreach ($config['resources'] as $name => $parameters) {
+            $this->addRequiredArgumentsToForms($name, $parameters, $container);
+        }
+
+        foreach ($this->reviewSubjects as $subject) {
+            $this->addProperTagToReviewDeleteListener($subject, $container);
+        }
+
+        $configFiles = [
+            'services.xml',
+        ];
+
+        foreach ($configFiles as $configFile) {
+            $loader->load($configFile);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function resolveResources(array $resources, ContainerBuilder $container)
+    {
+        $subjects = [];
+
+        foreach ($resources as $subject => $parameters) {
+            $this->reviewSubjects[] = $subject;
+            $subjects[$subject] = $parameters;
+        }
+
+        $container->setParameter('sylius.review.subjects', $subjects);
+
+        $resolvedResources = [];
+
+        foreach ($resources as $subjectName => $subjectConfig) {
+            foreach ($subjectConfig as $resourceName => $resourceConfig) {
+                if (is_array($resourceConfig)) {
+                    $resolvedResources[$subjectName.'_'.$resourceName] = $resourceConfig;
+                }
+            }
+        }
+
+        return $resolvedResources;
+    }
+
+    /**
+     * @param string           $name
+     * @param array            $parameters
+     * @param ContainerBuilder $container
+     */
+    private function addRequiredArgumentsToForms($name, array $parameters, ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('sylius.form.type.'.$name.'_review')) {
+            return;
+        }
+
+        foreach ($parameters['review']['classes']['form'] as $formName => $form) {
+            $formKey = ('default' === $formName) ? $name.'_review' : $name.'_review_'.$formName;
+            $formDefinition = $container->getDefinition('sylius.form.type.'.$formKey);
+            $formDefinition->addArgument($name);
+        }
+    }
+
+    /**
+     * @param string           $resourceName
+     * @param ContainerBuilder $container
+     */
+    private function addProperTagToReviewDeleteListener($resourceName, ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition('sylius.listener.review_delete')) {
+            return;
+        }
+
+        $listenerDefinition = $container->getDefinition('sylius.listener.review_delete');
+        $listenerDefinition->addTag('kernel.event_listener', ['event' => 'sylius.'.$resourceName.'_review.post_delete', 'method' => 'recalculateSubjectRating']);
```

Comment:
Smart, but we shouldn't depend on `sylius.xxx.post_delete` event since it will be invoked only when entity is removed through controller. E.g. I delete `Customer` and then his reviews are deleted by `cascade`, the rating will not be recalculated. Let's move it to doctrine `postRemove` event.